RFCs/0014-ToJsonString.md

```diff
@@ -0,0 +1,113 @@
+
+# Examples
+
+Here is an example of a resource containing a string-type property using a JSON formatted string.
+```json
+"MyDashboard": {
+    "Type": "AWS::CloudWatch::Dashboard",
+    "Properties": {
+        "DashboardBody": "{\"start\":\"-PT6H\",\"periodOverride\":\"inherit\",\"widgets\":[{\"type\":\"text\",\"x\":0,\"y\":7,\"width\":3,\"height\":3,\"properties\":{\"markdown\":\"Hello world\"}}]}"
+    }
+}
+```
+
+With the new `Fn::ToJsonString` intrinsic function we can simplify to the following:
+
+## JSON
+```json
+"MyDashboard": {
+    "Type": "AWS::CloudWatch::Dashboard",
+    "Properties": {
+        "DashboardBody": {
+            "Fn::ToJsonString": {
+                "start": "-PT6H",
+                "periodOverride": "inherit",
+                "widgets": [
+                    {  
+                        "type": "text",
+                        "x": 0,
+                        "y": 7,
+                        "width": 3,
+                        "height": 3,
+                        "properties": { "markdown": "Hello world" }
+                    }
+                ]
+            }
+        }
+    }
+}
+```
+
+## YAML
+```yaml
+MyDashboard:
+    Type: AWS::CloudWatch::Dashboard
+    Properties:
+        DashboardBody:
+            Fn::ToJsonString:
+                start: "-PT6H"
+                periodOverride: inherit
+                widgets:
+                    - type: text
+                    x: 0
+                    y: 7
+                    width: 3
+                    height: 3
+                    properties:
+                        markdown: Hello world
+```
+
+### Short Form Syntax
+```yaml
+MyDashboard:
+    Type: AWS::CloudWatch::Dashboard
+    Properties:
+        DashboardBody:
+            !ToJsonString:
+                start: "-PT6H"
+                periodOverride: inherit
+                widgets:
+                    - type: text
+                    x: 0
+                    y: 7
+                    width: 3
+                    height: 3
+                    properties:
+                        markdown: Hello world
+```
+
+# Motivation
+
+A CloudFormation user may want to use JSON strings as input to a resource property. For example, `AWS::CloudWatch::Dashboard` requires a JSON string for the `DashboardBody` attribute. A user may even want to use a JSON string as input to attributes where only a general string is required, such as in `SecretString` of `AWS::SecretsManager::Secret`.
+
+Today, this can only be accomplished through the use of external tools to convert object bodies to JSON strings, or use workarounds such as YAML multiline syntax. Having a feature to automate this transformation can provide various benefits, such as improving development workflow, improving code readability, or being able to utilize JSON/YAML syntax highlighting in text editors.
```

Comment:
I wonder if we should explain a bit more here what the benefit of this feature in Yaml templates is. (Yaml multiline is a nice workaround, but IDEs and text editors treat the content as generic text, not yaml, so there is generally no syntax highlighting or validation)

Reply:
I can add a bit more explaining this limitation with YAML multiline.